pandas/_libs/tslib.pyx

```diff
@@ -49,30 +49,31 @@ from pandas._libs.tslibs.tzconversion cimport (
 
 cdef inline object create_datetime_from_ts(
         int64_t value, npy_datetimestruct dts,
-        object tz, object freq):
+        object tz, object freq, bint fold):
     """ convenience routine to construct a datetime.datetime from its parts """
     return datetime(dts.year, dts.month, dts.day, dts.hour,
-                    dts.min, dts.sec, dts.us, tz)
+                    dts.min, dts.sec, dts.us, tz, fold=fold)
 
 
 cdef inline object create_date_from_ts(
         int64_t value, npy_datetimestruct dts,
-        object tz, object freq):
+        object tz, object freq, bint fold):
     """ convenience routine to construct a datetime.date from its parts """
+    # GH 25057 add fold argument to match other func_create signatures
     return date(dts.year, dts.month, dts.day)
 
 
 cdef inline object create_time_from_ts(
         int64_t value, npy_datetimestruct dts,
-        object tz, object freq):
+        object tz, object freq, bint fold):
```

Comment:
if adding `bint fold` caused a performance regression, might removing `freq` from these help get some of that back?

Reply:
I'll test, but I already tried decoupling these functionf from each other by dropping the `func_create` pointer, and there was no gain.

Reply:
@jbrockmendel 
Tried this with no effect (talking about Timestamp construction speed which we are worried about - not the performance of these specific functions):
```python
cdef inline object create_datetime_from_ts(
        int64_t value, npy_datetimestruct dts,
        object tz, object freq, bint fold):
    """ convenience routine to construct a datetime.datetime from its parts """
    cdef datetime dt
    dt = datetime_new(dts.year, dts.month, dts.day, dts.hour, dts.min, dts.sec,
                      dts.us, tz)
    dt = dt.replace(fold=fold)
    return dt


cdef inline object create_date_from_ts(
        int64_t value, npy_datetimestruct dts,
        object tz, object freq, bint fold):
    """ convenience routine to construct a datetime.date from its parts """
    # GH 25057 add fold argument to match other func_create signatures
    return date_new(dts.year, dts.month, dts.day)


cdef inline object create_time_from_ts(
        int64_t value, npy_datetimestruct dts,
        object tz, object freq, bint fold):
    """ convenience routine to construct a datetime.time from its parts """
    cdef time tm
    tm = time_new(dts.hour, dts.min, dts.sec, dts.us, tz)
    tm = tm.replace(fold=fold)
    return tm
```

As far as I can tell, regression to Timestamp construction speed was caused specifically by changing the signature of `create_timestamp_from_ts`. I also tried deleting the `func_create` pointer declaration, thus hopefully decoupling all the `func_create` candidates from each other, and it made no difference to performance, which supports this hypothesis.